### PR TITLE
boards: add nucleo-f042

### PR DESCRIPTION
**Pull request by vincent-d**
This PR adds the support for the tiny nucleo-f042 board. This board is based on a stm32f042k6 MCU.

I have succesfully ran hello_world, ipc_pingpong and default examples.

### CODE REVIEW

File: boards/nucleo-f042/include/periph_conf.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+/**
+ * @name RTC configuration
+ * @{
+ */
+/**
+ * Nucleos with MB1136 C-02 or MB1136 C-03 -sticker on it have the required LSE
+ * oscillator provided on the X2 slot.
+ * See Nucleo User Manual UM1724 section 5.6.2.
+ */
+//#define RTC_NUMOF           (1U)
```
Comment (aabadie) [inline boards/nucleo-f042/include/periph_conf.h]: RTC is supported on STM32f0 CPUs, so why commenting this out ?

**Comment by aabadie** [comment]
It would be great if you could add a wiki page [here](https://github.com/RIOT-OS/RIOT/wiki/RIOT-Platforms) describing this board.

File: boards/nucleo-f042/include/periph_conf.h (hunk 1 above)
Comment (vincent-d) [inline boards/nucleo-f042/include/periph_conf.h]: RTC is indeed supported, but there is no LSE on this board. I should change the define to 0, btw.

The other option is to add the capability of running with the LSI to the RTC driver, but this require more work ;)

**Comment by aabadie** [comment]
I ran a few tests (mainly the xtimer ones) and had some ram overflows at linkage. This CPU has only 6k of ram so you'll have to add it in the BOARD_INSUFFICIENT_MEMORY variable in a few Makefiles.
Let's ask Murdock about that.

**Comment by vincent-d** [comment]
@aabadie I am wondering if we could reduce the default thread stack size for this board. I ran some tests with the line `export CFLAGS=-DTHREAD_STACKSIZE_DEFAULT=512` in the board's Makefile.include and it run fine. Otherwise some tests just do not link or crash early when printing.

I'm not sure if it should be set on the board level or application level though.

**Comment by aabadie** [comment]
> @aabadie I am wondering if we could reduce the default thread stack size for this board. I ran some tests with the line export CFLAGS=-DTHREAD_STACKSIZE_DEFAULT=512 in the board's Makefile.include and it run fine.

Sounds reasonable. Maybe @OlegHahm, @kaspar030 have a better opinion on that.

**Comment by kYc0o** [comment]
I think the best place could be [here](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32f0/include/cpu_conf.h) just next to the header include.

**Comment by vincent-d** [comment]
board added to BAORD_INSUFFICIENT_MEMORY in tests which do not compile. Default stack size reduced in the way suggested by @kYc0o 

Rebased on latest master.

File: tests/conn_can/mapping_read.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import re
```
Comment (aabadie) [inline tests/conn_can/mapping_read.py]: Looks like you added this file by mistake (git add tests ?)

Comment (vincent-d) [inline tests/conn_can/mapping_read.py]: oops, indeed git add tests :D

**Comment by vincent-d** [comment]
fixed and squashed

**Comment by OlegHahm** [comment]
> @aabadie I am wondering if we could reduce the default thread stack size for this board. I ran some tests with the line export CFLAGS=-DTHREAD_STACKSIZE_DEFAULT=512 in the board's Makefile.include and it run fine. Otherwise some tests just do not link or crash early when printing.

Why do you think that this stack size would be sufficient for this board but not for other boards with the same MCU?

**Comment by vincent-d** [comment]
Well, I guess it depends on your application.

The stack size seems enough for basic application, but for bigger ones it could be an isssue. Anyway the RAM is small on this micro, so you will not be able to run a lot of threads, especially if you use some more memory for data. Moreover printf needs some free RAM used as heap.

I don't mind reverting back that change and letting the default stack size to 1024.

**Comment by OlegHahm** [comment]
I agree with your assessments, but I would leave it to the application developer to choose the optimized stack size values. The default values per MCU (family) are intended to be on the safe side for most standard cases.

**Comment by vincent-d** [comment]
I'm fine with that.

Stack size change removed, and squashed.

**Comment by OlegHahm** [comment]
In this case, immediate squashing is no issue, but usually you should not squash before a reviewer has approved the amendment (just in case you didn't know).

**Comment by aabadie** [comment]
@vincent-d, there are remaining issues reported by Murdock.

**Comment by vincent-d** [comment]
#6187 has been merged in between.

I will update that PR with the new uart driver.

**Comment by haukepetersen** [comment]
cool, thanks!

**Comment by vincent-d** [comment]
Do we care about the whitespace check fail in the vendor header ?

**Comment by OlegHahm** [comment]
If it's just failing because of trailing white spaces it should be super simple to fix, right?

**Comment by vincent-d** [comment]
Murdock still complains about some unrelated errors (lwip download failed ?) but it looks OK now.

Shall I squash ?

**Comment by aabadie** [comment]
can you:
```
git commit --amend
git push -f
``` 
?

**Comment by aabadie** [comment]
This will force a Murdock restart

**Comment by vincent-d** [comment]
Murdock seems happy.

**Comment by aabadie** [comment]
@vincent-d, Murdock is complaining because this branch needs squashing. Can you create 2 separate commits with comments:
* cpu/stm32f0: add support for stm32f042x6 mcu
* boards/nucleo-f042: initial support

**Pull request merged**